header/sync/sync.go

```diff
@@ -77,8 +85,13 @@ func (s *Syncer) Start(context.Context) error {
 
 // Stop stops Syncer.
 func (s *Syncer) Stop(context.Context) error {
-	s.cancel()
-	s.cancel = nil
+	if atomic.LoadUint64(&s.isRunning) == 1 {
+		// switch state to avoid getting in this statement twice
+		atomic.StoreUint64(&s.isRunning, 0)
```

Comment:
Btw, this could be on CompareAndSwap, but it is better to implement the suggestion above.